conda/mpich/meta.yaml

```diff
@@ -36,29 +36,26 @@ requirements:
     - mpich-mpicc {{ version }}
     - mpich-mpicxx {{ version }}
     - mpich-mpifort {{ version }}
-    - {{ moose_hdf5 }}
+    - hdf5=1.14.3=mpi*
     - {{ moose_krb5 }}
     - {{ moose_icu }}
     - {{ moose_libclang }}
     - {{ moose_libclang13 }}
     - {{ moose_clangcc }}       # [osx]
     - {{ moose_clangxx }}       # [osx]
-  host:
-    # TODO: Try to remove when this version settles down among other conda-forge packages
-    - openssl <3.2.0a           # [osx]
   run:
     - mpich {{ version }}
     - mpich-mpicc {{ version }}
     - mpich-mpicxx {{ version }}
     - mpich-mpifort {{ version }}
     - {{ moose_libgfortran }}
     - {{ moose_libgfortran5 }}
-    - {{ moose_hdf5 }}
+    - hdf5=1.14.3=mpi*
```

Comment:
This is now necessary. conda_build_config.yaml does not support variant wildcards (as far as I know).